test/langtools/tools/javac/T8271079.java

```diff
@@ -0,0 +1,101 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8271079
+ * @summary JavaFileObject#toUri in MR-JAR returns real path
+ * @modules java.compiler
+ *          jdk.compiler
+ * @run main T8271079
+ */
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import java.util.*;
+import java.util.jar.JarEntry;
+import javax.tools.*;
+
+public class T8271079 {
+
+  public static void main(String[] args) throws Exception {
+    Path mr = generateMultiReleaseJar();
+    try {
+      testT8271079(mr);
+    } finally {
+      Files.deleteIfExists(mr);
+    }
+  }
+
+  // $ echo 'module hello {}' > module-info.java
+  // $ javac -d classes --release 9 module-info.java
+  // $ jar --create --file mr.jar --release 9 -C classes .
+  private static Path generateMultiReleaseJar() throws Exception {
+    Files.writeString(Path.of("module-info.java"), "module hello {}");
+    java.util.spi.ToolProvider.findFirst("javac").orElseThrow()
+        .run(System.out, System.err, "-d", "classes", "--release", "9", "module-info.java");
+    Path mr = Path.of("mr.jar");
+    java.util.spi.ToolProvider.findFirst("jar").orElseThrow()
+        .run(System.out, System.err, "--create", "--file", mr.toString(), "--release", "9", "-C", "classes", ".");
+    System.out.println("Created: " + mr.toUri());
+    System.out.println(" Exists: " + Files.exists(mr));
+    return mr;
+  }
+
+  private static void testT8271079(Path path) throws Exception {
+    StandardJavaFileManager fileManager =
+        ToolProvider.getSystemJavaCompiler()
+            .getStandardFileManager(null, Locale.ENGLISH, StandardCharsets.UTF_8);
+    fileManager.setLocationFromPaths(StandardLocation.CLASS_PATH, List.of(path));
+    Iterator<String> options = Arrays.asList("--multi-release", "9").iterator();
+    fileManager.handleOption(options.next(), options);
+
+    Iterable<JavaFileObject> list =
+        fileManager.list(
+            StandardLocation.CLASS_PATH, "", EnumSet.allOf(JavaFileObject.Kind.class), false);
+
+    for (JavaFileObject f : list) {
+      System.out.println("JavaFileObject: " + f.getName());
+      System.out.println("JavaFileObject#toUri: " + f.toUri());
+      openUsingUri(f.toUri());
+    }
+    System.gc(); // JDK-8224794
+  }
+
+  private static void openUsingUri(URI uri) throws IOException {
+    URLConnection connection = uri.toURL().openConnection();
+    if (connection instanceof JarURLConnection) {
+      connection.setUseCaches(false); // JDK-8224794
+      try {
+        JarEntry entry = ((JarURLConnection) connection).getJarEntry();
+        System.out.println("JarEntry: " + entry.getName());
+        connection.getInputStream().close(); // JDK-8224794
+      } catch (FileNotFoundException e) {
+        e.printStackTrace();
```

Comment:
`printStackTrace` uses `System.err`. jtreg keeps output from `System.out` and `System.err` separate.  You  might want to consider using just one of the stream, not both.

It is common style in LangTools tests to set a member variable for the output stream, e.g. `PrintStream out;` so that it is a single assignment to declare use of `System.out` and `System.err`. It's also more concise at the use-sites. Generally, `System.err` is preferred for tracing/logging/error messages.